### DISCUSSION TITLE
Respect prefers-reduced-motion

### DISCUSSION THREAD

**Discussion by williewillus** [Able-Bear-Software/tab-nodes-tree#78] [Ideas / Feature Request]
The extension's animations for e.g. opening and closing new tabs doesn't respect the OS-level reduced-motion setting, accessible in CSS via the [prefers-reduced-motion](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) media query.

List of animations:
* Tab opening
* Tab closing
* Moving tabs around
* Initial tree load/reload
* Fading effect between selected tabs
* Fading from loading icon to site favicon
* Fading text when tab title changes

I don't like long drawn-out animations which is the main motivation for this feedback, but I imagine for other users this is an actual accessibility problem. Thanks!

**Comment by zeroarst** [comment]
Interesting. I didn't know `prefers-reduced-motion`. This will require some work since some of them are animated by dynamic style values not css. 
[Ticket created.](https://github.com/Able-Bear-Software/tab-nodes-tree/issues/79)

**Reply by williewillus** [reply]
fwiw you can call media queries like prefers-reduced-motion from JS as well using `window.matchMedia` https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia
